test/analysis/weightFunction.test.ts

```diff
@@ -86,12 +91,47 @@ describe('Weight Function correctly parses Argument Nodes if', () => {
         expect(getQueryTypeComplexity(queryAST, {}, customTypeWeights)).toBe(10);
     });
 
+    test('a custom object weight was set to 0', () => {
+        const customTypeWeights: TypeWeightObject = buildTypeWeightsFromSchema(schema, {
+            object: 0,
+        });
+        const query = `query { heroes(episode: NEWHOPE, first: 3) { stars, episode } }`;
+        const queryAST: DocumentNode = parse(query);
+        expect(getQueryTypeComplexity(queryAST, {}, customTypeWeights)).toBe(4); // 1 query and 4 enums
```

Comment:
Isn't the cost of resolving an enum 0? Should the expected result be 1 (q query + 3 * (0 + 0) heroes/character)?

Reply:
`Heros` is an object and there's 3 of them?

Reply:
Ok you are right, there was a bug that made the code default to the `DEFAULT_OBJECT_WEIGHT` when a custom weight of `0` is set because  `0` is falsy. I created a new Type called `TypeWeightsSet` which is the same as `TypeWeightsConfig` expect all the fields are required so we can eliminate the redundant `|| DEFAULT_OBJECT_WEIGHT` since the type weights are set at the start.

Reply:
Perfect, this makes much more sense than the original implementation.